packages/core/content-manager/admin/src/pages/EditView/components/DocumentActions.tsx

```diff
@@ -875,13 +1049,109 @@ const UpdateAction: DocumentActionComponent = ({
         );
 
         if ('data' in res && collectionType !== SINGLE_TYPES) {
-          navigate(
-            {
-              pathname: `../${res.data.documentId}`,
-              search: rawQuery,
-            },
-            { replace: true, relative: 'path' }
-          );
+          if (fromRelationModal) {
+            const createdRelation = {
+              documentId: res.data.documentId,
+              collectionType,
+              model,
+              params: currentDocumentMeta.params,
+            };
+            /*
+             * Update, if needed, the parent relation with the newly published document.
+             * Check if in history we have the parent relation otherwise use the
+             * rootDocument
+             */
+            if (
+              fieldToConnect &&
+              documentHistory &&
+              (parentDocumentMetaToUpdate.documentId ||
+                parentDocumentMetaToUpdate.collectionType === SINGLE_TYPES)
+            ) {
+              const parentDataToUpdate =
+                parentDocumentMetaToUpdate.collectionType === SINGLE_TYPES
+                  ? getInitialFormValues()
+                  : parentDocumentData.getInitialFormValues();
+
+              /*
+               * Check if the fieldToConnect is already present in the parentDataToUpdate.
+               * This happens in particular when in the parentDocument you have created
+               * a new component without saving.
+               */
+              const isFieldPresent = !!get(parentDataToUpdate, fieldToConnect);
+              const fieldToConnectPath = isFieldPresent
+                ? fieldToConnect
+                : // Compute the path to the parent object
+                  fieldToConnect.split('.').slice(0, -1).join('.');
+              const fieldToConnectValue = isFieldPresent
+                ? {
+                    connect: [
+                      {
+                        id: res.data.documentId,
+                        documentId: res.data.documentId,
+                        locale: res.data.locale,
+                      },
+                    ],
+                  }
+                : {
+                    [fieldToConnect.split('.').pop()!]: {
+                      connect: [
+                        {
+                          id: res.data.documentId,
+                          documentId: res.data.documentId,
+                          locale: res.data.locale,
+                        },
+                      ],
+                      disconnect: [],
+                    },
+                    __component: fieldToConnectUID,
+                  };
+              // In case the object was not present you need to pass the componentUID of the parent document
+              const objectToConnect = set({}, fieldToConnectPath, fieldToConnectValue);
+              const dataToUpdate = merge(parentDataToUpdate, objectToConnect);
+
```

Comment:
This same logic is used in two places. I think it should be a function with a signature like:

`connectRelationToParent(parentDataToUpdate, fieldToConnect, { data: res.data })`